src/app/model/Rock.ts

```diff
@@ -0,0 +1,109 @@
+import { Board } from './Board';
+import { PieceMovement } from './PieceMovement';
+import { Side, Square } from './Types';
+
+export class Rock implements PieceMovement {
+  id: string;
+  side: Side;
+
+  constructor(id: string, side: Side) {
```

Comment:
Dałbyś radę jakoś skrócić ten zapis? Skoro przypisujesz w konstruktorze pola to używając możesz to zrobić.

Reply:
dobre, nawet nie wiedziałem, że można to skrócić :-)